datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProviderAdmin.java

```diff
@@ -245,47 +243,90 @@ public void dropDestination(Datastream datastream) {
+        return Duration.ofMillis(Long.parseLong(entry.value()));
+      }
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.warn("Failed to retrieve config for topic {}.", topicName, e);
     }
-    return Duration.ofMillis(Long.parseLong(props.getProperty(TOPIC_RETENTION_MS)));
+
+    return null;
   }
 
   /**
    * Create Kafka topic based on the destination connection string, if it does not already exist.
    * @param connectionString connection string from which to obtain topic name
    * @param numberOfPartitions number of partitions
-   * @param topicConfig topic config to use for topic creation
+   * @param topicProperties topic config to use for topic creation
    */
-  public void createTopic(String connectionString, int numberOfPartitions, Properties topicConfig) {
+  public void createTopic(String connectionString, int numberOfPartitions, Properties topicProperties, Datastream datastream) {
     Validate.notNull(connectionString, "destination should not be null");
-    Validate.notNull(topicConfig, "topicConfig should not be null");
-    Validate.isTrue(_zkUtils.isPresent(), "zkUtils should be present");
+    Validate.notNull(topicProperties, "topicConfig should not be null");
 
     String topicName = KafkaTransportProviderUtils.getTopicName(connectionString);
-    populateTopicConfig(topicConfig);
-    try {
-      // Create only if it doesn't exist.
-      if (!AdminUtils.topicExists(_zkUtils.get(), topicName)) {
-        int replicationFactor = Integer.parseInt(topicConfig.getProperty("replicationFactor", DEFAULT_REPLICATION_FACTOR));
-        LOG.info("Creating topic with name {} partitions={} with properties {}", topicName, numberOfPartitions,
-                topicConfig);
+    populateTopicConfig(topicProperties);
 
-        AdminUtils.createTopic(_zkUtils.get(), topicName, numberOfPartitions, replicationFactor, topicConfig, RackAwareMode.Disabled$.MODULE$);
-      } else {
+    try {
+      AdminClient adminClient = getAdminClient(datastream);
+      NewTopic newTopic = getTopic(topicName, numberOfPartitions, topicProperties);
+      // Removing from topic config since its passed as a direct argument
+      topicProperties.remove("replicationFactor");
+      // TopicExistsException is thrown if topic exists. Its a no-op.
+      adminClient.createTopics(Collections.singletonList(newTopic)).all().get();
+    } catch (InterruptedException | ExecutionException e) {
+      if (e instanceof TopicExistsException || e.getCause() instanceof TopicExistsException) {
         LOG.warn("Topic with name {} already exists", topicName);
+      } else {
+        LOG.error(String.format("Creating topic %s failed with exception: ", topicName), e);
+        throw new DatastreamRuntimeException(e);
       }
     } catch (Throwable e) {
       LOG.error("Creating topic {} failed with exception {}", topicName, e);
       throw e;
     }
   }
 
+  private AdminClient getAdminClient(Datastream datastream) {
```

Comment:
Do we need to cache AdminClient instances based on the destinationBrokers ? Depends on how frequently this method is called and how expensive it is to re-instantiate AdminClient object, but I am not sure of it.